packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx

```diff
@@ -5,42 +5,50 @@ import { useStyle } from '../../../helpers';
 import type { AppearanceKey } from '../../../types';
 import { Root } from '../../elements';
 import { useTooltip } from './TooltipRoot';
+import { Motion } from '../Motion';
 
 export const tooltipContentVariants = () =>
   'nt-bg-foreground nt-p-2 nt-shadow-tooltip nt-rounded-lg nt-text-background nt-text-xs';
 
+type TooltipContentProps = JSX.IntrinsicElements['div'] & {
+  appearanceKey?: AppearanceKey;
+};
+
 const TooltipContentBody = (props: TooltipContentProps) => {
-  const { open, setFloating, floating, floatingStyles } = useTooltip();
+  const { open, setFloating, floating, floatingStyles, effectiveAnimationDuration } = useTooltip();
   const { setActive, removeActive } = useFocusManager();
   const [local, rest] = splitProps(props, ['class', 'appearanceKey', 'style']);
   const style = useStyle();
 
   onMount(() => {
     const floatingEl = floating();
-    setActive(floatingEl!);
+    if (floatingEl) setActive(floatingEl);
 
     onCleanup(() => {
-      removeActive(floatingEl!);
+      if (floatingEl) removeActive(floatingEl);
     });
   });
 
   return (
-    <div
+    <Motion.div
+      initial={{ opacity: 0, scale: 0.95 }}
+      animate={open() ? { opacity: 1, scale: 1 } : { opacity: 0, scale: 0.95 }}
+      transition={{ duration: effectiveAnimationDuration(), easing: 'ease-in-out' }}
```

Comment:
Remind me, why is this change neededD?

Reply:
because now it's much smoother 😃
this is the result of this motion, we were missing this effect before as well, check the action buttons.
it looks very strange on "Keyless mode" hover.
https://www.loom.com/share/74beed57dca2468e951c00e98beb7c07

https://www.loom.com/share/5c705e453b214aeeb1dbb355bfa5784e